core/src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -63,7 +81,41 @@ public ClientBuilder(String uri) {
      * Creates a new {@link ClientBuilder} that builds the client that connects to the specified {@link URI}.
      */
     public ClientBuilder(URI uri) {
-        this.uri = requireNonNull(uri, "uri");
+        this(requireNonNull(uri, "uri"), null, null, null);
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@code Scheme}.
+     */
+    public ClientBuilder(String scheme, Endpoint endpoint) {
+        this(Scheme.parse(scheme), requireNonNull(endpoint, "endpoint"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@link Scheme}.
+     */
+    public ClientBuilder(Scheme scheme, Endpoint endpoint) {
+        this(null, requireNonNull(scheme, "scheme"), null, requireNonNull(endpoint, "endpoint"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@link SessionProtocol}.
+     */
+    public ClientBuilder(SessionProtocol protocol, Endpoint endpoint) {
+        this(null, null, requireNonNull(protocol, "sessionProtocol"), requireNonNull(endpoint, "endpoint"));
+    }
+
+    private ClientBuilder(@Nullable URI uri, @Nullable Scheme scheme, @Nullable SessionProtocol protocol,
+                          @Nullable Endpoint endpoint) {
+        this.uri = uri;
+        this.scheme = scheme;
+        this.protocol = protocol;
+        this.endpoint = endpoint;
+
+        checkArguments();
```

Comment:
Seems that `check'Arguments'()` throws `IllegalStateException` not `IllegalArgumentExcpetion`.
I think we do not have to call it here and just check the state in `build()`

Reply:
Yes, I think so too.
Thanks!